### ISSUE TITLE
Doesn't error when passed a single blessed object

### ISSUE THREAD

**Issue by 2shortplanks** [houseabsolute/Params-CheckCompiler#3]
I don't expect a blessed object to be treated as a hashref

This test:

``` perl
use strict;
use warnings;

use Test2::Bundle::Extended;
use Test2::Plugin::NoWarnings;

use Params::CheckCompiler qw( compile );
use Types::Standard qw( Int );

{
    my $sub = compile(
        params => {
            foo => 1,
        },
    );

    is(
        dies { $sub->({ foo => 42 }) },
        undef,
        'lives when passed a hashref',
    );

    like(
        dies { $sub->(bless { foo => 42 }, 'anything') },
        qr/Expected a hash or hash reference but got a single blessed object argument/,
        'dies when passed a blessed object',
    );

}

done_testing();
```

Fails

```
Mark@travis:~/co/Params-CheckCompiler [master] $ perl -Ilib t/named/blessed.t
# Seeded srand with seed '20160606' from local date.
ok 1 - lives when passed a hashref
not ok 2 - dies when passed a blessed object
# Failed test 'dies when passed a blessed object'
# at t/named/blessed.t line 25.
# +---------+----+----------------------------------------------------------------------------+
# | GOT     | OP | CHECK                                                                      |
# +---------+----+----------------------------------------------------------------------------+
# | <UNDEF> | =~ | (?^:Expected a hash or hash reference but got a single blessed object argu |
# |         |    | ment)                                                                      |
# +---------+----+----------------------------------------------------------------------------+
1..2
```

**Comment by autarch** [comment]
Definitely a bug. I can implement a simple check for this in the generated code.

**Comment by autarch** [comment]
Fixed in 0.07.
